src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx

```diff
@@ -1,201 +1,234 @@
+      JSON.stringify({
+        problem_group: problemGroupValues || [],
+      }),
+    )
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [cachedValues])
+
+  const handleChangeProblemGroups = (
+    e: ChangeEvent<HTMLInputElement>,
+    problemGroup: string,
+  ) => {
+    const problemGroups = e.target.checked
+      ? [...problemGroupValues, problemGroup]
+      : problemGroupValues.filter((group: string) => group !== problemGroup)
+
+    handleSetConditions({ problemGroups })
+  }
+
+  const handleChangeVariants = (
+    e: ChangeEvent<HTMLInputElement>,
+    variantName: string,
+  ) => {
+    const variants = e.target.checked
+      ? [...variantsValues, variantName]
+      : variantsValues.filter(
+          (variantValue: string) => variantValue !== variantName,
+        )
+
+    handleSetConditions({ variants })
+  }
+
+  // cach values in every change
+  const handleSetConditions = ({
+    problemGroups,
+    variants,
+  }: {
+    problemGroups?: string[]
+    variants?: string[]
+  }): void => {
+    functionPanelStore.setCachedValues<IInheritanceModeCachedValues>(
+      FuncStepTypesEnum.InheritanceMode,
+      {
+        conditions: {
+          problem_group: problemGroups ? problemGroups : problemGroupValues,
+        },
+        variants: variants ? variants : variantsValues,
+      },
+    )
+
+    problemGroups &&
+      functionPanelStore.fetchStatFunc(
         FuncStepTypesEnum.InheritanceMode,
+        JSON.stringify({
+          problem_group: problemGroups || problemGroupValues,
+        }),
       )
+  }
+
+  const handleSumbitCondtionsAsync = async () => {
+    if (datasetStore.activePreset) datasetStore.resetActivePreset()
+
+    const inhModeConditions: TFuncCondition = [
+      'func',
+      FuncStepTypesEnum.InheritanceMode,
+      ConditionJoinMode.OR,
+      variantsValues,
+      {
+        problem_group:
+          problemGroupValues.length > 0 ? problemGroupValues : null,
+      },
+    ]
 
-    const [variants, setVariants] = useState([])
-    const [problemGroups, setProblemGroups] = useState<string[]>([])
+    const variant: TVariant = [`${variantsValues}`, 0]
 
-    const [problemGroupValues, variantsValues] = [
-      cachedValues?.problemGroups || [],
-      cachedValues?.variants || [],
-    ]
+    functionPanelStore.handleSumbitConditions(inhModeConditions, variant)
 
-    const fetchStatFuncAsync = async (
-      param?: Record<string, string | string[]>,
-    ) => {
-      const statFuncData = await filterStore.fetchStatFuncAsync(
-        FuncStepTypesEnum.InheritanceMode,
-        JSON.stringify(param) || JSON.stringify({ problem_group: [] }),
-      )
+    functionPanelStore.fetchStatFunc(
+      FuncStepTypesEnum.InheritanceMode,
+      JSON.stringify({ problem_group: problemGroupValues }),
+    )
+  }
 
-      const filteredVaraints = statFuncData?.variants?.filter(
-        item => item[1] > 0,
-      )
+  const handleSelectAllVariant = () => {
```

Comment:
draw you attention on how might it be simplified in this component. Maybe some functions are better to be moved in the store. In my perspective, there's too much logic